Lesson15/duygu/script.js

```diff
@@ -0,0 +1,631 @@
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word
+     - Joins them back
+   - Log the transformed string.
+*/
+function titleCase(sentence) {
+  let words = sentence.split(" ");
+  let capitalizedWords = words.map(word => 
+    word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
+  );
+  let transformedSentence = capitalizedWords.join(" ");
+
+  console.log(transformedSentence);
+  return transformedSentence;
+}
+
+
+titleCase("hello world");
+titleCase("this is a test");
+titleCase("javaScript IS fun!");
+
+
+/*
+20. Switch: Traffic Light
+   - Define a function `trafficLight(color)` that uses a switch statement:
+     - "red" -> log: "Stop"
+     - "yellow" -> log: "Caution"
+     - "green" -> log: "Go"
+     - anything else -> "Invalid color"
+*/
+function trafficLight(color) {
+  switch (color) {
+    case "red":
+      console.log("Stop");
+      break;
+    case "yellow":
+      console.log("Caution");
+      break;
+    case "green":
+      console.log("Go");
+      break;
+    default:
+      console.log("Invalid color");
+  }
+}
+
+trafficLight("red");
+trafficLight("yellow");
+trafficLight("green");
+trafficLight("Red");
+trafficLight("BLUE");
+
+/*
+21. Check String Length (if-else)
+   - Define a function `isLongString(str)` that checks if the string length
+     is more than 10.
+   - Log "Long string" or "Short string".
+*/
+function isLongString(str) {
+  if (str.length > 10) {
+    console.log("Long string");
+  } else {
+    console.log("Short string");
+  }
+}
+
+isLongString("Hello world!");
+isLongString("Hi!");
+
+
+/*
+22. Convert to Lowercase Then Check
+   - Define a function `isSpam(text)` that converts the text to lowercase
+     and checks if it includes "spam".
+   - If it does, log "This text is spam."
+   - Otherwise, log "This text is not spam."
+*/
+function isSpam(text) {
+  const checkedText = text.toLowerCase().includes("spam");
+  if (checkedText) {
+    console.log("This text is spam.");
+  } else {
+    console.log("This text is not spam.");
+  }
+}
+isSpam("PLEASE CHECK YOUR SPAM MAIL");
+isSpam("HI, HOW ARE YOU?");
+
+
+/*
+23. Extract Initials
+   - Define a function `getInitials(fullName)` that uses .split() to get each name part,
+     then logs the capitalized first letter of each.
+   - Example: "John Doe" -> "J.D."
+*/
+function getInitials(fullName) {
+  let seperated = fullName.split(" ").map(name => name[0].toUpperCase()).join(".");
+  return seperated;
+}
+
+console.log(getInitials("Süreyya Ekin Şen"));
+console.log(getInitials("İskender Yalım Şen"));
+/*
+24. Switch: Month to Season
+   - Define a function `getSeason(monthNum)` (1-12). Use switch or if-else:
+     - 12, 1, 2  -> "Winter"
+     - 3, 4, 5   -> "Spring"
+     - 6, 7, 8   -> "Summer"
+     - 9, 10, 11 -> "Autumn"
+   - Log the season or "Invalid month" if out of range.
+*/
+function getSeason(monthNum) {
+  switch (true) {
+    case monthNum === 12 || monthNum === 1 || monthNum === 2:
```

Comment:
```suggestion
    case 12:
    case 1:
    case 2:
      console.log("Winter");
      break;
```

Reply:
This is how you'd need to structure the switch case in this situation :) Update the rest.